exit.go

```diff
@@ -39,27 +39,40 @@ type ExitConfiguration struct {
 	SubscriptionFee      string                     `json:"SubscriptionFee"`
 	ClaimInterval        uint32                     `json:"ClaimInterval"`
 	Services             map[string]ExitServiceInfo `json:"Services"`
+	ReverseNanoPayFee    string                     `json:"ReverseNanopayfee"`
 }
 
 type TunaExit struct {
-	config           *ExitConfiguration
-	wallet           *nkn.Wallet
-	services         []Service
-	serviceConn      *cache.Cache
-	common           *Common
-	tcpListener      net.Listener
-	udpConn          *net.UDPConn
-	reverseIp        net.IP
-	reverseTcp       []int
-	reverseUdp       []int
-	onEntryConnected func()
-	closeChan        chan struct{}
+	config              *ExitConfiguration
+	Wallet              *nkn.Wallet
+	services            []Service
+	serviceConn         *cache.Cache
+	Common              *Common
+	tcpListener         net.Listener
+	udpConn             *net.UDPConn
+	reverseIp           net.IP
+	reverseTcp          []int
+	reverseUdp          []int
+	onEntryConnected    func()
+	closeChan           chan struct{}
+	reverseBytesIn      uint64
+	reverseBytesOut     uint64
+	reverseBytesInPaid  uint64
+	reverseBytesOutPaid uint64
 }
 
-func NewTunaExit(config *ExitConfiguration, services []Service, wallet *nkn.Wallet) *TunaExit {
+func NewTunaExit(config *ExitConfiguration, services []Service, reverseEntryToExitPrice, reverseExitToEntryPrice common.Fixed64, wallet *nkn.Wallet) *TunaExit {
 	return &TunaExit{
+		Common: &Common{
+			entryToExitPrice:   reverseEntryToExitPrice,
+			exitToEntryPrice:   reverseExitToEntryPrice,
```

Comment:
This should be `EntryToExitMaxPrice` and `ExitToEntryMaxPrice` instead of `entryToExitPrice` and `exitToEntryPrice`.
Note that max price and price are different:
Max price is user's max acceptable price (upper limit).
price is provider's asking price (actual price).